Only compress responses if request was compressed (#36867)
elastic/elasticsearch: docs/reference/modules/transport.asciidoc

cause idle connections to be closed, or by setting `transport.ping_schedule` if
keepalives cannot be configured.

[float]
==== Transport Compression

> By default Elasticsearch network-level compression is disabled. This default

Reviewer: ```suggestion
By default, network-level compression is disabled in Elasticsearch. This default
```

normally makes sense for local cluster communication as compression has a

> noticeable CPU cost and local clusters tend to be setup with fast network

Reviewer: ```suggestion
noticeable CPU cost and local clusters tend to be set up with fast network
```

> connections between nodes. If compression is necessary for remote cluster

Reviewer: ```suggestion
connections between nodes. If you want to compress remote cluster
```

> connections only, this can be configured on a per-cluster basis using the

Reviewer: The "only" here seems to imply that setting transport.compress affects both internode and remote cluster communications. i.e. you set this cluster-level setting if you *only* want to compress remote cluster connections.  If that's not right, we should remove the "only".

> `cluster.remote.${cluster_alias}.transport.compress` setting.

Reviewer: ```suggestion
<<remote-cluster-settings,`cluster.remote.${cluster_alias}.transport.compress` setting>>.
```

When compression is enabled using either `transport.compress` or for a specific

> remote cluster, only the outbound requests are compressed. Elasticsearch only

Reviewer 2: `Elasticsearch only` -> `Elasticsearch`

> remote cluster, only the outbound requests are compressed. Elasticsearch compresses

Reviewer: ```suggestion
remote cluster, only the outbound requests are compressed. 
```

> responses if and only-if the inbound request received was compressed.

Reviewer: ```suggestion
If compression is enabled and an inbound request is compressed, {es} also compresses the response.
```

Reviewer: Not sure if it's true that the compression of responses requires compression to be enabled, but it seemed like a reasonable assumption.
